build/source/netcdf/modelwrite.f90

```diff
@@ -263,11 +264,11 @@ subroutine writeBasin(modelTimestep,outputTimestep,meta,stat,dat,indx,err,messag
      selectcase (meta(iVar)%varType)
 
       case (iLookVarType%scalarv)
-       err = nf90_put_var(ncid(iFreq),meta(iVar)%ncVarID(iStat),(/stat(iVar)%dat(iStat)/),start=(/outputTimestep(iFreq)/),count=(/1/))
+       err = nf90_put_var(ncid(iFreq),meta(iVar)%ncVarID(iStat),(/stat(map(iVar))%dat(iStat)/),start=(/outputTimestep(iFreq)/),count=(/1/))
 
       case (iLookVarType%routing)
        if (modelTimestep==1) then
-        err = nf90_put_var(ncid(iFreq),meta(iVar)%ncVarID(iStat),(/dat(iVar)%dat/),start=(/1/),count=(/1000/))
+        err = nf90_put_var(ncid(iFreq),meta(iVar)%ncVarID(iStat),(/dat(map(iVar))%dat/),start=(/1/),count=(/1000/))
```

Comment:
It appears to be a hard upper limit on the number of days in the future that current rainfall *could* manifest as future runoff via a gamma routing function. This could go away if I ever took the time to figure out how to use ragged arrays in netcdf. Please notice that this hard limit appears in the `master` branch, and is not something that was added in any recent PR.